File: app/src/main/java/com/zhoujiulong/simpleprojectframework/base/BaseModel.java
package com.zhoujiulong.simpleprojectframework.base;

import com.zhoujiulong.baselib.http.HttpUtil;

public abstract class BaseModel<T> {

    /**
     * 網絡請求標記tag
     */
    protected String ReTag = getClass().getName();

    protected T mService;

    public BaseModel() {
        initService();
    }

    public abstract void initService();

    public void attachView(String reTag) {
        this.ReTag = reTag;
    }

    public void detachView() {
        HttpUtil.cancelWithTag(ReTag);
        if (mService != null) {
            mService = null;
        }
    }

}

File: app/src/main/java/com/zhoujiulong/simpleprojectframework/base/BaseModel.kt
package com.zhoujiulong.simpleprojectframework.base

import com.zhoujiulong.baselib.http.HttpUtil

abstract class BaseModel<T> {

    /**
     * 網絡請求標記tag
     */
    protected var ReTag = javaClass.name

    protected var mService: T

    init {
        mService = this.initService()
    }

    abstract fun initService() : T

    fun attachView(reTag: String) {
        this.ReTag = reTag
    }

    fun detachView() {
        HttpUtil.cancelWithTag(ReTag)
    }

}
